### PR TITLE
Bumped KeyVault versions for July 2020 patch release.

### PR DESCRIPTION
**Pull request by vcolin7**
- Bumped versions.
- Updated CHANGELOG and README.

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-secrets/pom.xml [hunk 1]
```diff
@@ -79,14 +79,14 @@
...
       <artifactId>azure-core-test</artifactId>
-      <version>1.3.0</version> <!-- {x-version-update;com.azure:azure-core-test;dependency} -->
+      <version>1.3.1</version> <!-- {x-version-update;com.azure:azure-core-test;dependency} -->
       <scope>test</scope>
     </dependency>
 
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-identity</artifactId>
-      <version>1.0.7</version> <!-- {x-version-update;com.azure:azure-identity;dependency} -->
```
Comment (samvaity) [inline sdk/keyvault/azure-security-keyvault-secrets/pom.xml:89]: Ideally, these changes should not have been a part of this PR.
cc: @JimSuplizio

Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-secrets/pom.xml:89]: So we keep changes like these when merging from master and only bump versions we control (e.g. KeyVault), right?

Comment (samvaity) [reply sdk/keyvault/azure-security-keyvault-secrets/pom.xml:89]: yes, sorry missed this was the hotfix branch.

**Comment by vcolin7** [comment]
@JimSuplizio Since this branch is for a stable release, shouldn't it use the latest stable core? I am under the impression that beta releases use the beta version of core.

**Comment by vcolin7** [comment]
Got the latest changes from master onto the release branch. cc @JimSuplizio @samvaity :)

**Pull request merged**